### PR TITLE
QL - Fix exception in job.get_job_ptr due to wrong string handling

### PR DESCRIPTION
**Pull request by rfehren**
This fix is necessary at least for Python3

### CODE REVIEW

File: pyslurm/pyslurm.pyx [hunk 1]
```diff
@@ -2160,7 +2160,7 @@ cdef class job:
             Job_dict[u'start_time'] = self._record.start_time
 
             if self._record.state_desc:
-                Job_dict[u'state_reason'] = self._record.state_desc.replace(" ", "_")
+                Job_dict[u'state_reason'] = self._record.state_desc.decode().replace(" ", "_")
```
Comment (giovtorres) [inline pyslurm/pyslurm.pyx]: Shouldn't this be `decode("utf-8")` or `decode("utf-8", "replace")` to avoid ascii conversion errors?

Comment (rfehren) [inline pyslurm/pyslurm.pyx]: In theory yes. In practice I'd consider it a bug, if slurm did supply non-ASCII characters in a job field. It's up to you.

Comment (rfehren) [inline pyslurm/pyslurm.pyx]: to be more precise: In a job field, where only a pre-defined set of values defined by slurm are possible. The job name e.g. can of course easily contain non-ASCII characters.

Comment (giovtorres) [inline pyslurm/pyslurm.pyx]: Yes. That's right.  I was thinking of a user-defined field.  Thanks!

**Pull request merged**